### ISSUE TITLE
BulkInsert - Is there an option to insert child records while using BulkInsert

### ISSUE THREAD

**Issue by vmohan074** [mikependon/RepoDB#699]
I am using RepoDB.BulkInsert(<Entity>) method, the child entities inside the entity object is not getting loaded to the database. Do we need to provide any option to achieve that?

**Comment by mikependon** [comment]
You are correct, RepoDB does not support JOIN, Relationships and Deep Object Graphs. You need to manually deal with that. We suggest you do the following.

First, bulk insert the parent data entity list with `isReturnIdentity` equals to `true` (see the explanation [here](https://repodb.net/operation/bulkinsert#special-arguments)). This will return the generated identity from the DB back to the parent data entities. Then, iterate each item in the collection and set the child entities entity based on the identity generated back from the database.

Then, do bulk insert all the child items together. During this time, ensure that all the child items has the parent id set already.
